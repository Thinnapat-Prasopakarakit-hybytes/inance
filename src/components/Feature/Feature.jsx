import React from "react";
import "./Feature.scss";
import { ReactComponent as Tools } from "../../assets/images/tools.svg";
import { ReactComponent as Construction } from "../../assets/images/construction.svg";
import { ReactComponent as Maintenance } from "../../assets/images/maintenance.svg";
const Feature = () => {
  return (
    <section className="feature_section">
      <div className="container">
        <div className="feature_container">
          <div className="box">
            <div className="img-box" data-testid="tools-icon">
              <Tools />
            </div>
            <h5 className="name">Repair</h5>
          </div>
          <div className="box active" data-testid="active">
            <div className="img-box" data-testid="construction-icon">
              <Construction />
            </div>
            <h5 className="name">Improve</h5>
          </div>
          <div className="box">
            <div className="img-box" data-testid="maintenance-icon">
              <Maintenance />
            </div>
            <h5 className="name">Maintain</h5>
          </div>
        </div>
      </div>
    </section>
  );
};

export default Feature;
