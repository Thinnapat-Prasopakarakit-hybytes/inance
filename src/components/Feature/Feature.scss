@use "../../styles/partials/variables" as *;
@use "../../styles/partials/mixins" as *;

.feature_section {
  transform: translateY(-50%);
  padding: 0;
  @include desktop {
    transform: none;
    padding: 90px 0;
  }

  .feature_container {
    display: flex;
    flex-direction: row;

    @include tablet {
      flex-direction: column;
    }

    .box {
      flex: 1;
      display: flex;
      flex-direction: column;
      align-items: center;
      text-align: center;
      margin: 0 10px;
      padding: 45px 15px;
      background-color: $white;
      color: #555089;
      transition: all 0.3s;
      box-shadow: 0 0 15px 0 rgba($color: $black, $alpha: 0.2);

      @include desktop {
        padding: 30px 15px;
      }

      &:not(:nth-last-child(1)) {
        margin-bottom: 0;
        @include tablet {
          margin-bottom: 15px;
        }
      }

      .img-box {
        height: 90px;
        width: 90px;
        @include desktop {
          width: 75px;
        }

        svg {
          width: 100%;
          height: auto;
          max-height: 100%;
          fill: #726dae;
          transition: all 0.3s;

          path {
            fill: #726dae;
          }
        }
      }

      .name {
        margin-top: 20px;
        text-transform: uppercase;
        @include secondary-font();
        margin-bottom: 0;
      }

      &:hover,
      &.active {
        background-color: $primary2;
        color: $white;

        .img-box {
          svg {
            fill: $white;

            path {
              fill: $white;
            }
          }
        }
      }
    }
  }
}
