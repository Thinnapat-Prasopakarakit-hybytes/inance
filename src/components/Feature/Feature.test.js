import React from "react";
import { render, screen } from "@testing-library/react";
import Feature from "./Feature";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";

jest.mock("../../assets/images/tools.svg", () => ({
  ReactComponent: () => <svg />,
}));

jest.mock("../../assets/images/construction.svg", () => ({
  ReactComponent: () => <svg />,
}));

jest.mock("../../assets/images/maintenance.svg", () => ({
  ReactComponent: () => <svg />,
}));

const renderFeatureComponent = () => {
  return render(
    <BrowserRouter>
      <Feature />
    </BrowserRouter>
  );
};

describe("Feature Component", () => {
  it("should render all feature boxes", () => {
    renderFeatureComponent();
    const boxes = screen.getAllByRole("heading");
    expect(boxes).toHaveLength(3);
  });

  it("should render all feature titles", () => {
    renderFeatureComponent();
    expect(screen.getByText("Repair")).toBeInTheDocument();
    expect(screen.getByText("Improve")).toBeInTheDocument();
    expect(screen.getByText("Maintain")).toBeInTheDocument();
  });

  it("should render all icon containers", () => {
    renderFeatureComponent();
    expect(screen.getByTestId("tools-icon")).toBeInTheDocument();
    expect(screen.getByTestId("construction-icon")).toBeInTheDocument();
    expect(screen.getByTestId("maintenance-icon")).toBeInTheDocument();
  });

  it("should have 'active' class on the Improve feature", () => {
    renderFeatureComponent();
    const improveBox = screen.getByTestId("active");
    expect(improveBox).toHaveClass("active");
  });

  it("should match snapshot", () => {
    const { asFragment } = renderFeatureComponent();
    expect(asFragment()).toMatchSnapshot();
  });
});
