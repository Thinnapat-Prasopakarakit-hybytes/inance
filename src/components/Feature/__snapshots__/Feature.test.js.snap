// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature Component should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="feature_section"
  >
    <div
      class="container"
    >
      <div
        class="feature_container"
      >
        <div
          class="box"
        >
          <div
            class="img-box"
            data-testid="tools-icon"
          >
            <svg />
          </div>
          <h5
            class="name"
          >
            Repair
          </h5>
        </div>
        <div
          class="box active"
          data-testid="active"
        >
          <div
            class="img-box"
            data-testid="construction-icon"
          >
            <svg />
          </div>
          <h5
            class="name"
          >
            Improve
          </h5>
        </div>
        <div
          class="box"
        >
          <div
            class="img-box"
            data-testid="maintenance-icon"
          >
            <svg />
          </div>
          <h5
            class="name"
          >
            Maintain
          </h5>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
